src/snowflake/snowpark/mock/_functions.py

```diff
@@ -787,7 +839,7 @@ def mock_to_timestamp_tz(
         sf_type=ColumnType(
             TimestampType(TimestampTimeZone.TZ), column.sf_type.nullable
         ),
-        dtype=column.dtype,
+        dtype=object,  # dtype being object to support VariantType
```

Comment:
Could you please elaborate on why the dtype being object helps with VariantType?

Reply:
The "support for VariantType" doesn't stand any more -- I find that set `dtype=column.dtype` will also pass all the tests.

this is more for:
1. alignment with other to_timestamp implementation
2. explicit casting dtype to object for internal calculation result storage -- imagine if the passed `column` dtype is str, then the dtype of the output result dataframe will also be set to string if code is written `dtype=column.dtype`, which converts all datetime object into string values inside the pandas dataframe.

Reply:
Is there no corresponding pandas dtype for `TimestampTtype`?

Reply:
there is the `numpy.datetime64`, yeah I would say ideally that type should be used here, but you remind me of one restriction I hit when fixing dtype drift: https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/mock/_util.py#L206-L222. I explicitly cast dtype to object for timestamp type to workaround the time range restriction